src/network/clientpackethandler.cpp

```diff
@@ -217,9 +217,28 @@ void Client::handleCommand_AccessDenied(NetworkPacket* pkt)
 		*pkt >> denyCode;
 		if (denyCode == SERVER_ACCESSDENIED_CUSTOM_STRING) {
 			*pkt >> m_access_denied_reason;
-		}
-		else if (denyCode < SERVER_ACCESSDENIED_MAX) {
-			m_access_denied_reason = accessDeniedStrings[denyCode];
+		} else {
+			if (denyCode < SERVER_ACCESSDENIED_MAX) {
+				m_access_denied_reason = accessDeniedStrings[denyCode];
+				if (denyCode == SERVER_ACCESSDENIED_SHUTDOWN ||
+						denyCode == SERVER_ACCESSDENIED_CRASH) {
+					*pkt >> m_access_denied_reason;
+				} else {
+					std::string reason_str;
+					*pkt >> reason_str;
+				}
+
+				u8 reconnect = 0;
+				*pkt >> reconnect;
+				m_access_denied_reconnect = reconnect & 1;
+			} else {
+				// Future proof-ness: allow us to add new error messages to the
```

Comment:
Definitively no, you do exactly same as denyCode == SERVER_ACCESSDENIED_CUSTOM_STRING here. Please remove this else.